src/components/lib/sync/sync_lock.h

```diff
@@ -73,6 +73,9 @@ sync_lock_teardown(struct sync_lock *l)
 static inline void
 sync_lock_take(struct sync_lock *l)
 {
+#if 1
```

Comment:
What is this? We are not suppose to have this code in the main line. Either change it to some macro which makes sense or delete the #else part.

Reply:
Also, this function generates a ton of warnings during the compilation.

Reply:
We really cannot add warnings. No code we add (aside from e.g. external libraries) should add warnings.